### PR TITLE
Adjust Transaction entity

### PR DESCRIPTION
**Pull request by ashkarpetin**
Added missing requestId property
Added support for alternative payment methods

### CODE REVIEW

**Review by colinbird — APPROVED** [review]

File: src/Entities/PaymentInstruments/AlternativeInstrument.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        parent::__construct($data);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function methodName()
+    {
+        return $this->alternativeMethod;
```
Comment (slavcodev) [inline src/Entities/PaymentInstruments/AlternativeInstrument.php]:
```suggestion
        return $this->getAttribute('method');
```

File: src/Entities/PaymentInstruments/AlternativeInstrument.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+use DomainException;
+use Rebilly\Entities\PaymentMethodInstrument;
+
+/**
+ * Class AlternativeInstrument
+ */
+class AlternativeInstrument extends PaymentMethodInstrument
+{
+    private $alternativeMethod;
```
Comment (slavcodev) [inline src/Entities/PaymentInstruments/AlternativeInstrument.php]:
```suggestion
```

File: src/Entities/PaymentInstruments/AlternativeInstrument.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (slavcodev) [inline src/Entities/PaymentInstruments/AlternativeInstrument.php]:
```suggestion
```

File: src/Entities/PaymentInstruments/AlternativeInstrument.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public const MSG_REQUIRED_METHOD = 'Method is required';
+
+    public function __construct(array $data = [])
+    {
+        if (!isset($data['method'])) {
+            throw new DomainException(self::MSG_REQUIRED_METHOD);
+        }
+
+        $this->alternativeMethod = $data['method'];
```
Comment (slavcodev) [inline src/Entities/PaymentInstruments/AlternativeInstrument.php]:
```suggestion
```

File: src/Entities/PaymentInstruments/AlternativeInstrument.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (slavcodev) [inline src/Entities/PaymentInstruments/AlternativeInstrument.php]:
```suggestion
```

File: src/Entities/PaymentInstruments/AlternativeInstrument.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        parent::__construct($data);
+    }
+
+    /**
+     * {@inheritdoc}
```
Comment (slavcodev) [inline src/Entities/PaymentInstruments/AlternativeInstrument.php]: No need

**Review by slavcodev — APPROVED** [review]

**Pull request merged**